### ISSUE TITLE
Bucket names are not validated

### ISSUE THREAD

**Issue by nbugden** [terraform-google-modules/terraform-google-cloud-storage#307]
### TL;DR

Terraform produces a valid plan despite bucket names not adhering to the [bucket name requirements in the docs](https://cloud.google.com/storage/docs/buckets#naming). When this plan is applied it fails to provision the bucket. 

### Expected behavior

Terraform plan should fail when the bucket name(s) are invalid. When validating the terraform plan as part of PR check requirements, this would block merging bucket name(s) that cannot be applied.

### Observed behavior

Terraform plan is successful even though bucket name(s) are invalid. When validating the terraform plan as part of PR check requirements, this would block allow merging bucket name(s) that cannot be applied. Resulting in a fix needing to be pushed.

### Terraform Configuration

```hcl
module "bucket" {
  source  = "terraform-google-modules/cloud-storage/google//modules/simple_bucket"
  version = "~> 5.0"

  name       = "my-really-long-bucket-name-with-invalid-ChAraCTers-*&^-and-google-in-the-name"
  project_id = "example-project"
  location   = "us-east1"
  iam_members = [{
    role   = "roles/storage.objectViewer"
    member = "user:example-user@example.com"
  }]
}
```

### Terraform Version

```sh
Terraform v1.7.4
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v5.22.0
+ provider registry.terraform.io/hashicorp/google-beta v5.22.0
+ provider registry.terraform.io/hashicorp/null v3.2.2
+ provider registry.terraform.io/hashicorp/random v3.6.0
+ provider registry.terraform.io/hashicorp/time v0.11.1
```

**Comment by nbugden** [comment]
Validate after the following changes are merged to the provider: \
- https://github.com/hashicorp/terraform-provider-google/issues/17831
- https://github.com/GoogleCloudPlatform/magic-modules/pull/10426
